### DISCUSSION TITLE
How github user can use 2FA if he don't use mobile apps and his country not in list?

### DISCUSSION THREAD

**Discussion by mobile-sergey** [community/community#63931] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

You wrote:
"GitHub users are [now required](https://github.blog/2023-03-09-raising-the-bar-for-software-security-github-2fa-begins-march-13/) to enable two-factor authentication as an additional security measure. Your activity on GitHub includes you in this requirement. You will need to enable two-factor authentication on your account before September 28, 2023, or be restricted from account actions."

But how user can set 2FA if he don't use mobile apps and his country not in [list](https://docs.github.com/ru/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported) (for example: Russia, Ukraine, Brazilia)?

<img width="619" alt="Снимок экрана 2023-08-16 в 15 25 56" src="https://github.com/community/community/assets/1146694/44fc0de3-793c-40d4-9e32-ce83d5c921f8">

<img width="619" alt="Снимок экрана 2023-08-16 в 15 26 07" src="https://github.com/community/community/assets/1146694/d6d5c0ae-afd1-4e01-ae4d-18e351b7bc44">

**Comment by oftharsh** [comment]
I understand your frustration, and I'm here to help clarify the situation. While GitHub does recommend using a mobile app as one of the methods for setting up two-factor authentication (2FA), it's not the only option available. GitHub provides several methods for setting up 2FA, and users who don't use mobile apps or are located in countries where certain methods may not be available can still enable 2FA using other methods.

Here are some alternative methods for setting up two-factor authentication on GitHub:

1. **Authentication Apps**: While mobile apps are commonly used, you can also use desktop-based authentication apps that generate time-based one-time passwords (TOTPs), such as Authy or WinAuth.

2. **SMS (Text Messages)**: You can receive authentication codes via SMS. This method might be available even in countries where mobile apps are not officially supported.

3. **Universal 2nd Factor (U2F) Keys**: Physical U2F keys like YubiKey can be used to enable 2FA. They provide an extra layer of security and don't require a mobile app.

4. **Recovery Codes**: GitHub provides recovery codes that you can use as an alternative to authentication apps or SMS. You can generate these codes and store them securely as a backup.

It's important to note that GitHub is continuously working to improve its security measures to protect users and their data. While these changes might bring some inconveniences, they are designed to enhance the overall security of the platform.

If you have specific concerns about setting up 2FA or encounter any difficulties due to your location or preferences, I recommend reaching out to GitHub's support or help center. They can provide guidance and assistance tailored to your situation.

Please keep in mind that security measures are crucial to maintaining the integrity of online platforms, and companies like GitHub aim to strike a balance between security and user experience.

**Reply by mobile-sergey** [reply]
Unfortunately If I push "Enable two-factor authorization"...

<img width="642" alt="Снимок экрана 2023-08-16 в 15 27 18" src="https://github.com/community/community/assets/1146694/5b1e449d-31ee-4c63-a7db-5c0df6868ebc">

I can use only 2 methods (screenshots of this 2 methods add to topic):
1) Authentication Apps
2) SMS (Text Messages)

No "Universal 2nd Factor (U2F) Keys" and "Recovery Codes"

**Comment by oftharsh** [comment]
I apologize for the confusion. If you don't have access to the methods you mentioned (U2F keys and recovery codes), you can still use one of the available options: Authentication Apps or SMS (Text Messages). Here's a brief overview of both methods:

1. **Authentication Apps**:
   Authentication apps like Authy, Google Authenticator, or Microsoft Authenticator can generate time-based one-time passwords (TOTPs) for 2FA. You'll need to install one of these apps on your device and then follow these steps:

   - Choose the "Authentication Apps" option when enabling 2FA.
   - Open the authentication app and scan the QR code displayed on GitHub.
   - The app will generate a time-based code. Enter this code on the GitHub website to complete the setup.

2. **SMS (Text Messages)**:
   While SMS-based 2FA is not the most secure option due to potential vulnerabilities, it's still an available choice:

   - Choose the "SMS (Text Messages)" option when enabling 2FA.
   - Enter your phone number, and GitHub will send a verification code via SMS.
   - Enter the verification code on the GitHub website to complete the setup.

Please note that SMS-based 2FA can be less secure because text messages can potentially be intercepted or accessed by attackers. If possible, I recommend using an authentication app for generating TOTP codes, as it provides a higher level of security.

If neither of these options is suitable for you, you may want to consider contacting GitHub Support directly to inquire about alternative options for enabling 2FA based on your specific circumstances and location. They may be able to provide further assistance or suggestions tailored to your situation.

**Comment by LOuroboros** [comment]
> You wrote: "GitHub users are [now required](https://github.blog/2023-03-09-raising-the-bar-for-software-security-github-2fa-begins-march-13/) to enable two-factor authentication as an additional security measure. Your activity on GitHub includes you in this requirement. You will need to enable two-factor authentication on your account before September 28, 2023, or be restricted from account actions."
> 
> But how user can set 2FA if he don't use mobile apps and his country not in [list](https://docs.github.com/ru/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported) (for example: Russia, Ukraine, Brazilia)?

I'd love to know about this too. Uruguay is also not in the list of acceptable countries to send an SMS to, but even if it was, to be forced to buy a phone just to avoid losing access to the page's features is ludicrous, imo.

There isn't even a page that explains how much accessibility will I lose if I refuse to enable 2FA.

Forcing the user to enable 2FA is annoying enough as is, but putting that aside, I am positively and 100% sure that it costs exactly nothing at all to provide an option like letting the user use their email address *(or a new, secondary address)* for the 2FA authentication, or use any other sensible solution.

**Comment by DNin01** [comment]
If you aren't able to download a mobile authenticator app, there are also authenticator apps that work on computers.
